src/titlepane/TitlePane.ts

```diff
@@ -0,0 +1,118 @@
+import uuid from '@dojo/core/uuid';
+import { v } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { theme, ThemeableMixin, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+
+import * as css from './styles/titlePane.m.css';
+
+/**
+ * @type TitlePaneProperties
+ *
+ * Properties that can be set on a TitlePane component
+ *
+ * @property closeable          If false the pane will not collapse in response to clicking the title
+ * @property headingLevel       'aria-heading-level' for the title's DOM node
+ * @property open               If true the pane is opened and content is visible
+ * @property title              Title to display above the content
+ * @property onRequestClose     Called when the title of an open pane is clicked
+ * @property onRequestOpen      Called when the title of a closed pane is clicked
+ */
+export interface TitlePaneProperties extends ThemeableProperties {
+	closeable?: boolean;
+	headingLevel?: number;
+	open?: boolean;
+	title: string;
+	onRequestClose?(titlePane: TitlePane): void;
+	onRequestOpen?(titlePane: TitlePane): void;
+};
+
+export const TitlePaneBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class TitlePane extends TitlePaneBase<TitlePaneProperties> {
+	private _afterRender(element: HTMLElement) {
+		// Conditionally adjust top margin. Done manually instead of through Maquette
+		// so the underlying DOM is accessible, as we need to know the content height.
+		// Put in a timeout to push this operation to the next tick, otherwise
+		// element.offsetHeight below can be incorrect (e.g. before styling is applied)
+		requestAnimationFrame(() => {
+			const { open = true } = this.properties;
+			const height = element.offsetHeight;
+			element.style.marginTop = open ? '0px' : `-${ height }px`;
```

Comment:
Is there a reason to animate `margin-top` instead of `height`? I had thought `height` would be more performant, but I haven't tested those two directly against each other.